### PR TITLE
docs: Add Limitations documentations

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/8409

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,12 @@ These resources can be helpful to refine your issue, work out an application/dri
     * [./tests](./tests) - Information about testing the Validation Layers.
     * [GOVERNANCE.md](GOVERNANCE.md) - Repository management details.
 
+### Internals
+
+The Validation Layers are broken up as many smaller internal "Validation Objects" each tasked with own job.
+[Core Checks](./docs/core_checks.md) is what is on by default, but there are other validation such as [Synchronization Validation](./docs/synchronization.md), [GPU Assisted Validation](./docs/gpu_validation.md), and [more](./docs/README.md).
```
Comment (ziga-lunarg) [inline README.md:42]: `what is on by default`, what is enabled by default might be better wording. 

Also `but there are other validation such as`, is there a word missing after `validation`?

File: docs/limitations.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+There are things that are invalid, that could only be caught by the source language.
+
+### pNext and sType
+
+If an app is trying to use `VkBindImageMemoryInfo`, but is setting the `sType` as `VK_STRUCTURE_TYPE_BIND_BUFFER_MEMORY_INFO` (mixed up the buffer and image name), this might cause issues.
+
+The Validation Layers (as well as any other layer or driver) uses the `sType` to know how to cast the `void* pNext`, so if it is wrong, there is no way a Vulkan Layer could know.
+
+There are VUID such as `VUID-VkBufferCreateInfo-pNext-pNext` that limit which `sType` can appear in the pNext chain, but something the 2 `sType` being mixed up both might be valid to use in that case.
```
Comment (ziga-lunarg) [inline docs/limitations.md:23]: I feel like the last part of the sentence here could be made clearer.

File: README.md (hunk 1 above)
Comment (spencer-lunarg) [reply README.md:42]:
```suggestion
[Core Checks](./docs/core_checks.md) is what is enabled by default, but there are other validation that can be additionally enabled, such as [Synchronization Validation](./docs/synchronization.md), [GPU Assisted Validation](./docs/gpu_validation.md), and [more](./docs/README.md).
```

File: docs/limitations.md (hunk 2 above)
Comment (spencer-lunarg) [reply docs/limitations.md:23]:
```suggestion
There are VUID such as `VUID-VkBufferCreateInfo-pNext-pNext` that limit which `sType` can appear in the pNext chain, but something the 2 mixed up `sType` can both be valid by chance, and there will be no validation error.
```

File: docs/limitations.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+While the Validation Layers thrive to catch all undefined behavior, because the checking is done at a Vulkan Layer level, not everything is possible to validate.
+
+This documentation is not served as an "errata" or a "list of all known limitations", but rather just a general outline so people are aware there are some limitations.
+
+## Unimplementable Validation
+
+The Validation Layers use the [VUID](https://github.com/KhronosGroup/Vulkan-Guide/blob/main/chapters/validation_overview.adoc#valid-usage-id-vuid) to know what validation is [covered](https://vulkan.lunarg.com/doc/sdk/latest/windows/validation_error_database.html). There are times, for various reasons, we are unable to validate the VUID, but don't want it to marked as "yet to do" as there is nothing for us to actually do. The [unimplementable_validation.h](../layers/error_message/unimplementable_validation.h) file was created to list those.
+
+This file breaks down all VUID with a reason why the Validation Layers are not able to actually validate them.
```
Comment (arno-lunarg) [inline docs/limitations.md:11]:
```suggestion
```

File: docs/limitations.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+## Source Code
+
+There are things that are invalid, that could only be caught by the source language.
```
Comment (arno-lunarg) [inline docs/limitations.md:15]:
```suggestion
There are things that are invalid, that can only be caught by the source language.
```

**Review by arno-lunarg — APPROVED** [review]

File: docs/limitations.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+There are things that are invalid, that can only be caught by the source language.
+
+### pNext and sType
+
+If an app is trying to use `VkBindImageMemoryInfo`, but is setting the `sType` as `VK_STRUCTURE_TYPE_BIND_BUFFER_MEMORY_INFO` (mixed up the buffer and image name), this might cause issues.
+
+The Validation Layers (as well as any other layer or driver) uses the `sType` to know how to cast the `void* pNext`, so if it is wrong, there is no way a Vulkan Layer could know.
+
+There are VUID such as `VUID-VkBufferCreateInfo-pNext-pNext` that limit which `sType` can appear in the pNext chain, but something the 2 mixed up `sType` can both be valid by chance, and there will be no validation error.
```
Comment (filnet) [inline docs/limitations.md:22]: something => sometimes ?

File: docs/limitations.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+### Dereferencing Pointers
+
+There are VUs that will be worded something like `must be a valid pointer to ...`.
+
+The Validation Layers will check the pointer is not null, but if the pointer is pointing to garbage, there is no way to safely dereference it.
```
Comment (filnet) [inline docs/limitations.md:28]: will check [that] the pointer

File: docs/limitations.md (hunk 5 above)
Comment (spencer-lunarg) [reply docs/limitations.md:22]: addressed in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/8440

**Pull request merged**